src/o365/graph/commands/teams/teams-channel-set.ts

```diff
@@ -0,0 +1,149 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import { GraphItemsListCommand } from '../GraphItemsListCommand';
+import Utils from '../../../../Utils';
+import * as request from 'request-promise-native';
+import { Channel } from './Channel';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+interface Options extends GlobalOptions {
+  teamId: string;
+  channelName: string;
+  newChannelName: string;
+  description: string
+}
+class GraphTeamsChannelSetCommand extends GraphItemsListCommand<Channel>{
+  public get name(): string {
+    return `${commands.TEAMS_CHANNEL_SET}`;
+  }
+  public get description(): string {
+    return 'Updates properties of a specified channel in the given Microsoft Teams team';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.role = args.options.teamId;
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    let channelId: string = '';
+    const endpoint: string = `${auth.service.resource}/v1.0/teams/${encodeURIComponent(args.options.teamId)}/channels`;
+    this.getAllItems(endpoint, cmd, true)
```

Comment:
Instead of retrieving all channels, we can find the one we need using the following call: `https://graph.microsoft.com/v1.0/teams/3afb0e04-8bad-4005-9e96-420d0d0dc42d/channels?$filter=displayName eq 'General'`